src/iris/utils.py

```diff
@@ -312,6 +312,66 @@ def claim_incidents_from_batch_id(batch_id, owner):
     connection.close()
 
 
+def resolve_incident(incident_id, owner):
+    connection = db.engine.raw_connection()
+
+    cursor = connection.cursor()
+    cursor.execute('''
+      SELECT `target`.`name`
+      FROM `incident`
+      LEFT JOIN `target` ON `target`.`id` = `incident`.`owner_id`
+      WHERE `incident`.`id` = %s
+    ''', (incident_id, ))
+    result = cursor.fetchone()
+    previous_owner = result[0] if result else None
+    cursor.close()
+
+    resolved = 1 if owner else 0
+
+    # if incident is unclaimed claim it when marking it as resolved
+    if resolved == 1:
+        claim_incident(incident_id, owner)
+    else:
+        owner = previous_owner
+
+    now = datetime.datetime.utcnow()
+
+    max_retries = 3
+
+    for i in xrange(max_retries):
+        cursor = connection.cursor()
+        try:
+            cursor.execute('''UPDATE `incident`
+                               SET `incident`.`updated` = %(updated)s,
+                                   `incident`.`resolved` = %(resolved)s,
+                                   `incident`.`owner_id` = (SELECT `target`.`id` FROM `target` WHERE `target`.`name` = %(owner)s AND `type_id` = (SELECT `id` FROM `target_type` WHERE `name` = 'user'))
```

Comment:
I don't think clicking "resolved" should change the incident owner. Owner should match claimer, and we can add a new field if we want to track resolver. We can get rid of the owner_id update this way.